ice/settings.py

```diff
@@ -1,22 +1,28 @@
+import json
+
 from os import environ
 from pathlib import Path
+from typing import Any
+from typing import Optional
 from typing import TYPE_CHECKING
 
 from pydantic import BaseSettings
 
 if TYPE_CHECKING:
     AnyHttpUrl = str
-
 else:
     from pydantic import AnyHttpUrl
 
+from .logging import log_lock
+
+OUGHT_ICE_DIR = Path(environ.get("OUGHT_ICE_DIR", Path.home() / ".ought-ice"))
+
+_env_path = OUGHT_ICE_DIR / ".env"
```

Comment:
I preferred the original position these lines were in.

Reply:
They have to be defined above `__get_and_store`, which uses them.

Reply:
No, they only have to be defined before `__get_and_store` is called.

Reply:
Ah whoops, thanks! Doing momentarily

Reply:
Btw, why do you prefer the original position? I would have guessed the opposite since I suspect that (eg) linters will complain about these variables being possibly undefined. (Maybe this is a gap in my Python tooling and runtime knowledge?)

Reply:
I'd be surprised if a linter would complain about this since the function using the outer (global in this case) variable might only be called long after the variable is defined, which is definitely the case here, and is probably even typical. Similarly, linters don't generally complain about the order in which functions are defined.

I prefer the position because it's closer to the 'primary' place where it's used, which is to initialise `settings`.